src/shared/tpm2-util.c

```diff
@@ -1457,6 +1457,31 @@ int tpm2_digest_many_digests(
         return tpm2_digest_many(alg, digest, iovecs, n_data, extend);
 }
 
+static int tpm2_set_auth(Tpm2Context *c, const Tpm2Handle *handle, const char *pin) {
+        TPM2B_AUTH auth = {};
+        TSS2_RC rc;
+        int r;
+
+        assert(c);
+        assert(handle);
+
+        if (!pin)
+                return 0;
+
+        CLEANUP_ERASE(auth);
+
+        r = tpm2_digest_buffer(TPM2_ALG_SHA256, &auth, pin, strlen(pin), /* extend= */ false);
+        if (r < 0)
+                return r;
```

Comment:
if you make tpm2_digest_buffer() non-logging, then i guess you need to add logging here

Reply:
changed tpm2_digest_many() to log at error, so tpm2_digest_buffer() will also